homeassistant/components/sonos/speaker.py

```diff
@@ -538,8 +537,25 @@ async def async_rebooted(self, soco: SoCo) -> None:
             self.zone_name,
             soco,
         )
-        await self.async_unseen(will_reconnect=True)
-        await self.async_seen(soco)
+        await self.async_unsubscribe()
+        self.soco = soco
+        await self.async_subscribe()
+        if self._seen_timer:
+            self._seen_timer()
+        self._seen_timer = self.hass.helpers.event.async_call_later(
+            SEEN_EXPIRE_TIME.total_seconds(), self.async_unseen
+        )
+        if self._poll_timer:
+            self._poll_timer()
+        self._poll_timer = self.hass.helpers.event.async_track_time_interval(
+            partial(
+                async_dispatcher_send,
+                self.hass,
+                f"{SONOS_POLL_UPDATE}-{self.soco.uid}",
+            ),
+            SCAN_INTERVAL,
+        )
```

Comment:
If the poll timer is already running we should be able to let it keep running. However if it does _not_ exist, it's a good idea to start a new one in case there are issues with the subscription callbacks. If the subscriptions work it will be cancelled during the first successful callback. Something like this:
```suggestion
        if not self._poll_timer:
            self._poll_timer = self.hass.helpers.event.async_track_time_interval(
                partial(
                    async_dispatcher_send,
                    self.hass,
                    f"{SONOS_POLL_UPDATE}-{self.soco.uid}",
                ),
                SCAN_INTERVAL,
            )
```

Reply:
Or perhaps it makes more sense to move the poll timer creation inside `async_subscribe()` instead. It's a safety fallback in case the subscription callbacks fail to arrive and should always be created in tandem with subscriptions.

Reply:
If there isn't a scenario where you might want to subscribe and NOT setup a poll timer then yeah it would be nicer to setup the poll timer inside `async_subscribe`

Reply:
Yes, I think they should be tied together.

However I think what you have here now is good and solves the problem at hand. Further refactoring to better organize the poll timer handling can be done in a followup PR.